backends/smt2/smtbmc.py

```diff
@@ -1596,6 +1634,10 @@ def smt_check_sat(expected=["sat", "unsat"]):
                 print_anyconsts(num_steps)
                 print_failed_asserts(num_steps)
                 write_trace(step, num_steps+1, '%', allregs=True)
+                if find_loops:
+                    loop = find_state_loop(step, num_steps+1)
+                    if loop:
+                        print_msg(f"Loop detected, increasing induction depth will not help. Cycle {loop[0]} = cycle {loop[1]}")
```

Comment:
smtbmc uses "step" instead of "cycle" for this. As a step doesn't have to correspond to a clock cycle (when using `clk2fflogic`) that's more precise.

Reply:
Implemented fully